internal/services/applications/applications.go

```diff
@@ -793,6 +853,26 @@ func flattenApplicationSpa(in *msgraph.ApplicationSpa) []map[string]interface{}
 	}}
 }
 
+func flattenApplicationPasswordCredentials(passwordCredentials *[]msgraph.PasswordCredential, d map[string]interface{}) []map[string]interface{} {
+	if d == nil {
+		return []map[string]interface{}{}
+	}
+
+	// update, read case
+	if d["key_id"] != nil {
+		if credRead := helpers.GetPasswordCredential(passwordCredentials, d["key_id"].(string)); credRead != nil {
+			return helpers.FlattenCredential(credRead, d)
+		}
+	}
+
+	// create case
+	for _, appCred := range *passwordCredentials {
+		return helpers.FlattenCredential(&appCred, d)
+	}
+
+	return []map[string]interface{}{}
+}
```

Comment:
By moving this logic into the resource for improved readability, we can simplify this function

```suggestion
func flattenApplicationPasswordCredentials(input *[]msgraph.PasswordCredential) (output []map[string]interface{}) {
	output = make([]map[string]interface{}, 0)

	if input == nil {
		return
	}

	for _, in := range *input {
		var startDate, endDate string

		if in.StartDateTime != nil {
			startDate = in.StartDateTime.Format(time.RFC3339)
		}
		if in.EndDateTime != nil {
			endDate = in.EndDateTime.Format(time.RFC3339)
		}

		output = append(output, map[string]interface{}{
			"key_id":       pointer.From(in.KeyId),
			"display_name": pointer.From(in.DisplayName),
			"start_date":   startDate,
			"end_date":     endDate,
			"value":        pointer.From(in.SecretText),
		})
	}

	return
}
```